crates/nu-command/src/generators/seq.rs

```diff
@@ -324,74 +326,87 @@ fn print_seq(
     pad: bool,
     padding: usize,
     span: Span,
+    engine_state: &EngineState,
 ) -> PipelineData {
     let mut i = 0isize;
     let mut value = first + i as f64 * step;
     // for string output
     let mut ret_str = "".to_owned();
-    // for number output
-    let mut ret_num = vec![];
     // If the separator and terminator are line endings we can convert to numbers
     let use_num =
         (separator == "\n" || separator == "\r") && (terminator == "\n" || terminator == "\r");
 
+    if use_num {
+        let int_stream: bool = first.fract() == 0.0 && step.fract() == 0.0;
```

Comment:
I don't want to trust float precision for any precise counting. We should specialize the implementation to either use a `Counter<i64>` or `Counter<f64>`.